### PR TITLE
chore: precompile with babel

### PR DESCRIPTION
**Pull request by SimenB**
Prerequisite for #256. This has no TS yet, but sets up babel for jest and publishing.

This also allows us to use newer syntax while maintaining node 6 compat

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -32,6 +32,8 @@ module.exports = {
       },
     ],
     'prettier/prettier': 'error',
+    'node/no-unsupported-features/es-syntax': 'off',
```
Comment (SimenB) [inline .eslintrc.js]: babel deals with syntax

File: package.json [hunk 2]
```diff
@@ -16,24 +16,31 @@
...
   ],
+  "main": "lib/",
   "engines": {
     "node": ">=6"
   },
   "peerDependencies": {
     "eslint": ">=5"
   },
   "scripts": {
+    "postinstall": "yarn build",
```
Comment (jeysal) [inline package.json]: Is auto-build after install helpful in this case?
Unlike in the Jest repo, you'll be able to run `yarn jest` just fine.
In development, `yarn build --watch` will initially build everything anyway (afaik that's what Babel CLI `watch` does)

File: babel.config.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+'use strict';
+
+module.exports = {
+  presets: [['@babel/preset-env', { targets: { node: 6 } }]],
```
[the remark this answers is not in the document]
Comment (SimenB) [reply babel.config.js]: No need to make a breaking change, exactly 0 benefit

File: package.json (hunk 2 above)
Comment (SimenB) [reply package.json]: Lint will fail since we run with our own rules. And jest runs lint, so it'd fail

Comment (jeysal) [reply package.json]: Okay, didn't consider that 👍

File: babel.config.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (SimenB) [reply babel.config.js]: whenever we wanna make a new major we can drop 6 though, no reason to keep it at that point

File: package.json (hunk 2 above)
Comment (ranyitz) [reply package.json]: @SimenB `postInstall` runs when this package is being `npm/yarn` installed.

We can't assume that the user has `yarn` or any of the `devDependencies`.

I think we should revert it for now.

Comment (ranyitz) [reply package.json]: Maybe the intention was `prepublishOnly`?

**Pull request merged**